Bugzilla/DaemonControl.pm

```diff
@@ -0,0 +1,327 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::DaemonControl;
+use 5.10.1;
+use strict;
+use warnings;
+
+use Bugzilla::Constants qw(bz_locations);
+use Cwd qw(realpath);
+use English qw(-no_match_vars $PROGRAM_NAME);
+use File::Spec::Functions qw(catfile);
+use Future::Utils qw(repeat try_repeat);
+use Future;
+use IO::Async::Loop;
+use IO::Async::Process;
+use IO::Async::Protocol::LineStream;
+use IO::Async::Signal;
+use IO::Socket;
+use LWP::Simple qw(get);
+use POSIX qw(setsid WEXITSTATUS);
```

Comment:
there's still modules missing from Makefile.PL – `Future`, and `IO::Async`.

Reply:
Future is a dep of io async and we list io-async as part lf elasticsearch. We don’t have optional dependencies in actuality anymore so this is fine.

Reply:
sorry, but i don't agree.

you're relying on modules that you depend on to pull in modules that you use directly.  this relies on those modules always using those modules, and removes clarity around which modules are required.

if you use a module directly it should be documented as a dependency.

Reply:
The chance of IO::Async not depending on Future is 0%. They're tightly integrated.

Reply:
I will however make sure this code checks for feature('bmo')

Reply:
as per irc, let's not block this any longer.  we can (and should!) add the deps later.